pkgs/development/python-modules/openstacksdk/default.nix

```
@@ -0,0 +1,84 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, appdirs
+, cryptography
+, ddt
+, dogpile_cache
+, hacking
+, jmespath
+, jsonpatch
+, jsonschema
+, keystoneauth1
+, munch
+, netifaces
+, os-service-types
+, oslo-config
+, oslotest
+, pbr
+, prometheus-client
+, requests-mock
+, requestsexceptions
+, stestr
+, testscenarios
+}:
+
+buildPythonPackage rec {
+  pname = "openstacksdk";
+  version = "0.59.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-PfdgzScjmKv6yM6+Yu64LLxJe7JdTdcHV290qM6avw0=";
+  };
+
+  propagatedBuildInputs = [
+    appdirs
+    cryptography
+    dogpile_cache
+    jmespath
+    jsonpatch
+    keystoneauth1
+    munch
+    netifaces
+    os-service-types
+    pbr
+    requestsexceptions
+  ];
+
+  checkInputs = [
+    ddt
+    hacking
+    jsonschema
+    oslo-config
+    oslotest
+    prometheus-client
+    requests-mock
+    stestr
+    testscenarios
+  ];
+
+  checkPhase = ''
+    stestr run -e <(echo "
+    openstack.tests.unit.cloud.test_image.TestImage.test_create_image_task
+    openstack.tests.unit.image.v2.test_proxy.TestImageProxy.test_wait_for_task_error_396
+    openstack.tests.unit.image.v2.test_proxy.TestImageProxy.test_wait_for_task_wait
```

This test was also failing on my machine

```suggestion
    openstack.tests.unit.image.v2.test_proxy.TestImageProxy.test_wait_for_task_wait
    openstack.tests.unit.test_resource.TestWaitForStatus.test_status_fails
```